document/CoreFormat.mdk

```diff
@@ -454,13 +463,21 @@ A `result` object describes a single result produced by a static analysis tool.
 
 ### `ruleId` property { #result-ruleId }
 
-A `result` object shall contain a property named `ruleId` whose value is a string containing
+A `result` object should contain a property named `ruleId` whose value is a string containing
 the stable, opaque identifier for the rule that was evaluated to produce the result.
 
 ~ IsoExample
 `"CA2101"`
 ~
 
+Static analysis tools that produce the SARIF format shall provide a `ruleId` for each `result`
+(with the possible exception of `result`s whose `kind` property is `note`; see (&sect;[#result-kind]).
+
+Not all existing static analysis tools emit the equivalent of a `ruleId` in their output.
+A conversion tool which converts the output of such a static analysis tool to the SARIF format
```

Comment:
@lgolding, I disagree, no synthesis. It is a driving principle for SARIF that converters should avoid trying to synthesize SARIF constructs where there is no original tool equivalent. Converters should populate SARIF constructs where tool equivalents exist, populate properties/tags with per-tool data for which there are no SARIF equivalents and stop there. If we encourage converters to normalize/synthesize values, this introduces new versioning issues in viewers (because a converter might make a bad choice synthesizing a value and fix this problem in an update). This creates additional complexity. We don't currently have a placeholder for converter version in the standard, by design. So, converters should put original tool data in the appropriate slot, if available, and then pass all remaining per-tool data through. A viewer can then examine the source tool and its version and update/refine its logic for synthesizing missing things, if required.